Set user for rkt tasks

chelseakomlo opened the pull request:
Further testing is necessary for each case

jippi commented on client/driver/docker.go:
```diff
@@ -201,6 +201,7 @@ type DockerDriverConfig struct {
 	MacAddress       string              `mapstructure:"mac_address"`        // Pin mac address to container
 	SecurityOpt      []string            `mapstructure:"security_opt"`       // Flags to pass directly to security-opt
 	Devices          []DockerDevice      `mapstructure:"devices"`            // To allow mounting USB or other serial control devices
+	User             string              `json:"User,omitempty" yaml:"User,omitempty" toml:"User,omitempty"`
```
whats up with the yaml and toml here?

chelseakomlo commented:
Testing suggestions are more than welcome.

dadgar commented on client/driver/docker.go:
```diff
@@ -201,6 +201,7 @@ type DockerDriverConfig struct {
 	MacAddress       string              `mapstructure:"mac_address"`        // Pin mac address to container
 	SecurityOpt      []string            `mapstructure:"security_opt"`       // Flags to pass directly to security-opt
 	Devices          []DockerDevice      `mapstructure:"devices"`            // To allow mounting USB or other serial control devices
+	User             string              `mapstructure:user`
```
Why add a new config field instead of using the task.User?

chelseakomlo commented:
Changed this as we already expose this functionality for Docker, see https://www.nomadproject.io/docs/job-specification/task.html#user. 

Added the change to pass this through for rkt.

dadgar commented on client/driver/rkt.go:
```diff
@@ -569,6 +569,11 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
 
 	}
 
+	// If a user has been specified for the task, pass it through to the user
```
See if you can add tests to this and for docker

dadgar commented on client/driver/rkt_test.go:
```diff
@@ -366,18 +368,19 @@ func TestRktDriverUser(t *testing.T) {
+
+	select {
+	case res := <-resp.Handle.WaitCh():
+		assert.False(res.Successful())
```
Can you assert it fails b/c the user
